pkg/autodiscovery/common/utils/prometheus_apiserver.go

```diff
@@ -25,14 +26,14 @@ const (
 )
 
 // ConfigsForService returns the openmetrics configurations for a given service if it matches the AD configuration
-func (pc *PrometheusCheck) ConfigsForService(svc *v1.Service) []integration.Config {
+func ConfigsForService(pc *types.PrometheusCheck, svc *v1.Service) []integration.Config {
 	var configs []integration.Config
 	namespacedName := fmt.Sprintf("%s/%s", svc.GetNamespace(), svc.GetName())
-	if pc.isExcluded(svc.GetAnnotations(), namespacedName) {
+	if pc.IsExcluded(svc.GetAnnotations(), namespacedName) {
 		return configs
 	}
 
-	instances, found := pc.buildInstances(svc.GetAnnotations(), namespacedName)
+	instances, found := BuildInstances(pc, svc.GetAnnotations(), namespacedName)
```

Comment:
It seems that `BuildInstances` can be private

Reply:
indeed, thank you for pointing out! change done